### DISCUSSION TITLE
New Feature: Executes commands in the shell when you close tabs or exit the Windows Terminal.

### DISCUSSION THREAD

**Discussion by fxzxmic** [microsoft/terminal#14546] [Ideas]
Generally, this command is "exit", this can trigger "bash_logout", or a custom command to wrap up before closing the shell.

**Comment by zadjii-msft** [comment]
This seems like it would be fraught with edge cases. Would the terminal just type some specified input into the connection, hit enter, and then close the tab? What if you're running some text editor instead of sitting at the shell prompt? Or are we literally `CreateProcess`'ing a new commandline?

Is there a particular bug you're trying to solve with this request? I think we've got an open issue for "closing a tab doesn't do a `bassh_logout`"...

**Reply by zadjii-msft** [reply]
er, https://github.com/microsoft/terminal/issues/7826#issuecomment-703808683 might have been what I was thinking of

**Reply by fxzxmic** [reply]
I think the specific implementation of this feature still needs to be discussed, so I just publish it as a discussion rather than an issue. Just take it as a flash of inspiration.
